### PR TITLE
Add Terms of Use acceptance page

### LINKED ISSUE #155: Add a modal dialog to make users agree to a privacy / data use policy
When a user first visits any page other than the home page or legal pages, they should be presented with a modal dialog, with the embedded site Terms of Use. In order to proceed, they will have to click to accept these policies. Choosing not to accept the policies should redirect back to the home page.

Once a user accepts the policy, this should be noted in either their user account (if logged in) or a session cookie (if anonymous), so the dialogs do not appear again.

### PR DESCRIPTION
**Pull request by brianhelba**
Fixes #155.

TODO:
- [x] Fix the style of the `termsAcceptancePage`.
- [x] Attempt to unify style between `termsAcceptancePage` and `uploadDataset`.
- [x] Consider storing the date of acceptance in the server's `user['acceptTerms']` field, instead of just `True`.
- [x] Store anonymous users acceptance in `localStorage` instead of `sessionStorage`. This makes debugging slightly more tedious, however.

### CODE REVIEW

**Comment by msmolens** [comment]
@brianhelba This should be complete besides the change of `sessionStorage' to `localStorage`. One thing we could change is the few places we refer to "Terms of Service"--that's inconsistent with "Terms of Use" used elsewhere.

**Comment by brianhelba** [comment]
@msmolens I've just updated 3 more things. Let me know if you approve (since I opened the PR, I need your review to merge).

File: web_external/Legal/TermsAcceptanceView.js [hunk 1]
```diff
@@ -5,15 +5,10 @@ isic.views.TermsAcceptanceView = isic.View.extend({
...
             buttons.prop('disabled', true);
 
-            if (girder.currentUser) {
-                girder.currentUser.setAcceptTerms(function () {
-                    // Refresh page
-                    Backbone.history.loadUrl();
-                });
-            } else {
-                window.sessionStorage.setItem('acceptTerms', 'true');
+            girder.models.UserModel.currentUserSetAcceptTerms(function () {
```
Comment (msmolens) [inline web_external/Legal/TermsAcceptanceView.js]: **isic**.models.UserModel

File: web_external/Legal/TermsAcceptanceView.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+isic.views.TermsAcceptanceView = isic.View.extend({
+    events: {
+        'click #isic-terms-accept': function (event) {
+            // Disable buttons while request is pending
+            var buttons = this.$('.isic-terms-agreement-button-container button');
+            buttons.prop('disabled', true);
+
+            girder.models.UserModel.currentUserSetAcceptTerms(function () {
```
Comment (msmolens) [inline web_external/Legal/TermsAcceptanceView.js]: Needs to be `isic.models.UserModel`.

**Pull request merged**